### PR TITLE
fix issues #3329 ,see this :http://www.cocos2d-x.org/issues/3329

### PR DESCRIPTION
**Pull request by justbilt**
fix issues #3329 ,see this :http://www.cocos2d-x.org/issues/3329

### CODE REVIEW

File: cocos/audio/win32/MciPlayer.cpp [hunk 1]
```diff
@@ -1,212 +1,108 @@
...
 
 MciPlayer::MciPlayer()
-: _wnd(NULL)
-, _dev(0L)
-, _soundID(0)
-, _times(0)
-, _playing(false)
+: m_nSoundID(0)
+, m_uTimes(0)
+, m_scr(NULL)
```
Comment (dumganhar) [inline cocos/audio/win32/MciPlayer.cpp]: Please don't use `Hungarian notation`. Please following https://github.com/cocos2d/cocos2d-x/blob/develop/docs/CODING_STYLE.md. Thanks.

File: cocos/audio/win32/SimpleAudioEngine.cpp [hunk 2]
```diff
@@ -123,15 +165,34 @@ bool SimpleAudioEngine::isBackgroundMusicPlaying()
...
-    if (p != sharedList().end())
-    {
-        p->second->Play((bLoop) ? -1 : 1);
-    }
+	if (!pszFilePath)
+	{
+		return 0;
+	}
+
+	//�������Ƿ��л���õ�
```
Comment (dumganhar) [inline cocos/audio/win32/SimpleAudioEngine.cpp]: Please use English only.

**Comment by justbilt** [comment]
1.delete chinese comment
2.remove Hungarian notation

File: cocos/audio/win32/MciPlayer.cpp [hunk 3]
```diff
@@ -1,212 +1,108 @@
...
-
-        _dev = mciOpen.wDeviceID;
-        _soundID = uId;
-        _playing = false;
-    } while (0);
+	_scr=_device->GetScreamer(pFileName);
+
+	if (_scr==NULL)
+	{
+		if (!cocos2d::CCFileUtils::getInstance()->isFileExist(_fileName))
```
Comment (WenhaiLin) [inline cocos/audio/win32/MciPlayer.cpp]: Is the parameter should be pFileName?

File: cocos/audio/win32/MciPlayer.cpp [hunk 4]
```diff
@@ -1,212 +1,108 @@
...
+		{
+			assert(_scr!=NULL && "File not exist!");
+		}
+		else
+		{
+			assert(_scr!=NULL && "Only suppost .ogg file!");
```
Comment (WenhaiLin) [inline cocos/audio/win32/MciPlayer.cpp:36]: I think we should use the openaudioengine after it support more format.

Comment (minggo) [inline cocos/audio/win32/MciPlayer.cpp:36]: It is another solution. Now we can just focus on this pull request.

File: cocos/audio/win32/MciPlayer.cpp (hunk 3 above)
Comment (minggo) [inline cocos/audio/win32/MciPlayer.cpp]: @justbilt 
Is it a mistake?

Comment (minggo) [inline cocos/audio/win32/MciPlayer.cpp]: @Dhilan007 
Please modify and send a new pull request.

File: cocos/audio/win32/MciPlayer.cpp [hunk 5]
```diff
@@ -1,212 +1,108 @@
...
-        mciOpen.lpstrElementName = pFileName;
-
-        mciError = mciSendCommand(0,MCI_OPEN, MCI_OPEN_ELEMENT, reinterpret_cast<DWORD_PTR>(&mciOpen));
-        BREAK_IF(mciError);
-
-        _dev = mciOpen.wDeviceID;
-        _soundID = uId;
-        _playing = false;
-    } while (0);
+	_scr=_device->GetScreamer(pFileName);
```
Comment (WenhaiLin) [inline cocos/audio/win32/MciPlayer.cpp:26]: It crash when call the Open method for same file again.You can test it  step by step: 
1.TestCpp-->CocosDenshionTest
  a.play music or effect
  b.back to  main menu
  c.enter into CocosDenshionTest
  d.play music or effect

File: cocos/audio/win32/MciPlayer.h [hunk 6]
```diff
@@ -14,56 +16,30 @@ class MciPlayer
...
     void Close();
 
-    /**
-    @brief 播放声音文件
-    @param pFileName 播放的声音文件名称，需要包含文件的路径
-    @param nTimes    播放声音文件的循环次数，默认值为 1，即播放一次
-    */
     void Open(const char* pFileName, UINT uId);
 
     void Play(UINT uTimes = 1);
```
Comment (WenhaiLin) [inline cocos/audio/win32/MciPlayer.h:21]: It's have not effect for call the method like this Play(-1) at the first time.The effect play just once.

**Comment by minggo** [comment]
@justbilt 
Thanks for you reply.
There are two issues:
1. Please don't use Chinese words in commit info
2. How about the problem the @Dhilan007 said?

**Comment by cocosQA** [comment]
Dear justbilt:
  We have created a new branch 'v3' to replace branch 'develop', and this PR has been moved from 'develop' branch to 'v3' branch, the new PR is https://github.com/cocos2d/cocos2d-x/pull/6471
The old 'develop' branch will be deleted soon, and this PR will be closed accordingly.

**Pull request closed without merging**